contributing/core/testing.md

```diff
@@ -0,0 +1,64 @@
+# Testing
+
+## Running tests
+
+We recommend running the tests in headless mode (with the browser windows hidden) and with a specific directory pattern and/or test name (`-t`) which ensures only a small part of the test suite is run locally:
+
+For example, running one test in the production test suite:
+
+````
+Running one test in the `test/integration/production` test suite:
+
+```sh
+pnpm test -- test/integration/production -t "should allow etag header support"
+````
+
+Running all tests in the `test/integration/production` test suite:
+
+```sh
+pnpm test test/integration/production
+```
+
+When you want to debug a particular test you can replace `pnpm test` with `pnpm testonly` to opt out of the headless browser.
+When the test runs it will open the browser that is in the background by default, allowing you to inspect what is on the screen.
```

Comment:
In order to manually inspect what is happening in the browser, most of the time I have to add an `await waitFor(300000)` before the first `expect`. And I have to remove it before I commit the change.

Maybe there should be a better way to do this?

Reply:
We would like to revisit our Next.js test utils later this year. :+1: